### PR TITLE
refactor: standardise on dispute_id hash

### PR DESCRIPTION
**Pull request by evilrobot-01**
Proposed changes to standardise on `dispute_id` as hash as per parachain contracts, with incrementing vote rounds per `dispute_id`.

### CODE REVIEW

**Comment by evilrobot-01** [comment]
@oraclown any idea offhand on the maximum number of voting rounds per `dispute_id` hash? Based on the increasing dispute fee, I figure a `u8` might suffice instead of a `u32`.

**Comment by oraclown** [comment]
> @oraclown any idea offhand on the maximum number of voting rounds per `dispute_id` hash? Based on the increasing dispute fee, I figure a `u8` might suffice instead of a `u32`.

Yes, `u8` or 255 vote rounds seems like a fine max value.

File: runtime-api/src/lib.rs [hunk 1]
```diff
@@ -266,15 +266,16 @@ sp_api::decl_runtime_apis! {
...
+		/// Determines if an account voted for a specific dispute round.
 		/// # Arguments
 		/// * `dispute_id` - The identifier of the dispute.
+		/// * `vote_round` - The vote round.
 		/// * `voter` - The account of the voter to check.
 		/// # Returns
-		/// Whether or not the account voted for the specific dispute.
-		fn did_vote(dispute_id: DisputeId, voter: AccountId) -> bool;
+		/// Whether or not the account voted for the specific dispute round.
+		fn did_vote(dispute_id: DisputeId, vote_round: u32, voter: AccountId) -> bool;
```
Comment (oraclown) [inline runtime-api/src/lib.rs:278]: Changing `vote_round` to `u8` I think per your last comment

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks @evilrobot-01 . Looks like the `u32`s need to be changed to `u8`, but can't spot anything else. The dispute id creation looks good, if `Token::FixedBytes` generates bytes32 equivalent

File: runtime-api/src/lib.rs (hunk 1 above)
Comment (evilrobot-01) [reply runtime-api/src/lib.rs]: Done

**Comment by evilrobot-01** [comment]
> The dispute id creation looks good, if `Token::FixedBytes` generates bytes32 equivalent

`Token::FixedBytes` is from https://docs.rs/ethabi/18.0.0/ethabi/enum.Token.html#variant.FixedBytes and is currently only used in tests to avoid taking a dependency on an external library within the pallet. Note that it is only added as a [dev dependency](https://github.com/tellor-io/tellor-pallet/blob/b3194f8cdf2a04c45240f5c8812393fa047506e3/Cargo.toml#L32).

https://github.com/tellor-io/tellor-pallet/blob/b3194f8cdf2a04c45240f5c8812393fa047506e3/src/contracts/mod.rs#L37 provides a simple implementation which is then compared with `Token::FixedBytes` in tests [(example](https://github.com/tellor-io/tellor-pallet/blob/44c41d9e5a5e18324d7c7a8d78a5bb1a0d3e07e3/src/contracts/mod.rs#L130)). 

We could also add a few additional tests which compare output from these functions with ABI encoded output from something like https://abi.hashex.org/ if it provides additional assurance?

**Review by oraclown — APPROVED** [review]
Thanks, @evilrobot-01

**Review by pawanbisht62 — APPROVED** [review]
LGTM

**Pull request merged**